libarchive/archive_read_support_filter_zstd.c

```diff
@@ -213,7 +213,9 @@ zstd_filter_read(struct archive_read_filter *self, const void **p)
 
 	state = (struct private_data *)self->data;
 
-	out = (ZSTD_outBuffer) { state->out_block, state->out_block_size, 0 };
+	out.dst = state->out_block;
+	out.pos = state->out_block_size;
+	out.size = 0;
```

Comment:
The original code will _always_ zero fill any struct fields that are not mentioned. I would use either `out = {}`, `out = { 0 }` or `memset()` - whichever is more common in the project.